tests/draft7/ref.json

```diff
@@ -1012,5 +1012,49 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "$ref uses base URI learned from $id",
+        "schema": {
+            "allOf": [
+                {"$ref": "#/$defs/referencing"}
+            ],
+            "$defs": {
```

Comment:
`$defs` should be `definitions` for draft 7

Reply:
(It's weird that the sanity check for unknown keywords doesn't flag this, but I don't have time to look into why right this second.)